conans/test/download_test.py

```diff
@@ -22,6 +24,59 @@ class HelloConan(ConanFile):
 
 class DownloadTest(unittest.TestCase):
 
+    def test_returns_on_failures(self):
+        test_server = TestServer([("*/*@*/*", "*")], [("*/*@*/*", "*")])
+        servers = {"default": test_server}
+        client2 = TestClient(servers=servers)
+        client2.init_dynamic_vars()
+
+        conan_ref = ConanFileReference.loads("Hello/1.2.1@frodo/stable")
+        package_ref = PackageReference(conan_ref, "123123123")
+
+        class Response(object):
+            ok = None
+            status_code = None
+            charset = None
+            text = ""
+
+            def __init__(self, ok, status_code):
+                self.ok = ok
+                self.status_code = status_code
+
+        class BuggyRequester(object):
+            def get(self, *args, **kwargs):
+                return Response(False, 404)
+
+        client2.remote_manager._remote_client._rest_client.requester = BuggyRequester()
```

Comment:
Not a fan about it, going to submit a pull request and check what do you think about my changes, better than explaining it here.

See [pull request](https://github.com/lasote/conan/pull/12)

CC: @memsharded